### DISCUSSION TITLE
Episode 131: One Algorithm To Rule Them All!

### DISCUSSION THREAD

**Discussion by codereport** [codereport/adsp2#22] [Episode]
Episode Link: https://adspthepodcast.com/2023/05/26/Episode-131.html

Feel free to write comments or ask questions :slightly_smiling_face:

**Comment by assafcohen** [comment]
Thank you for the great content 
Nice talk on cache was on cppcon
https://youtu.be/0iXRRCnurvo

**Comment by DanielAjoy** [comment]
Connor mentioned Brian Harvey in his SICP talk. Brian also uses the "cascade" higher order function to manage repeated application of functions with initial inputs. The description is here:

https://people.eecs.berkeley.edu/~bh/pdf/v1ch05.pdf

Here is the code for the Fibonacci sequence, two initial inputs and two different functions at play:

```
to fib :n
output (cascade :n [?1+?2] 1 [?1] 0)
end

print fib 5
8
print fib 23
46368
```

**Reply by codereport** [reply]
Interesting. Haven't heard of the name `cascade` for this yet. Thanks!

**Comment by vmiheer** [comment]
At the end of Episode, Bryce: Should've said for_each instead of reduce. And I am like okay we have come to that level of saying it should be `if_call` it's like `if_jump` (the heart of Turing machine/assembly) but for functional programmers :).
